src/OrleansRuntime/Streams/QueueBalancer/DeploymentBasedQueueBalancer.cs

```diff
@@ -98,19 +98,33 @@ public class DeploymentBasedQueueBalancer : QueueBalancerBase, ISiloStatusListen
             await NotifyListeners();
         }
 
+        private async Task HandleSiloStatusChange()
+        {
+            while (!this.siloStatusOracle.CurrentStatus.IsTerminating())
+            {
+                try
+                {
+                    this.siloStatusSharedState = await this.siloStatusSharedState.NextAsync;
+                    HandleSiloStatusChange(this.siloStatusSharedState.State);
```

Comment:
Regarding the performance, we'll always have a pending task on the caller thread (in this case, the queue manager's thread), but that does not take up cpu.